rotkehlchen/chain/ethereum/modules/scroll_bridge/decoder.py

```diff
@@ -0,0 +1,369 @@
+            return DEFAULT_DECODING_OUTPUT
+
+        from_address = hex_or_bytes_to_address(tx_log.topics[3])
+        to_address = hex_or_bytes_to_address(tx_log.data[:32])
+
+        if not self.base.any_tracked([from_address, to_address]):
+            return DEFAULT_DECODING_OUTPUT
+
+        ethereum_token_address = hex_or_bytes_to_address(tx_log.topics[1])
+        asset = self.base.get_or_create_evm_token(ethereum_token_address)
+        raw_amount = hex_or_bytes_to_int(tx_log.data[32:64])
+        amount = asset_normalized_value(raw_amount, asset)
+
+        expected_event_type, new_event_type, from_chain, to_chain, _ = bridge_prepare_data(
+            tx_log=tx_log,
+            deposit_topics=(DEPOSIT_ERC20,),
+            source_chain=ChainID.ETHEREUM,
+            target_chain=ChainID.SCROLL,
+            from_address=from_address,
+            to_address=to_address,
+        )
+
+        # Find the fee refund event and remove it
+        refund_event_idx, refund_amount = None, FVal(0)
+        for idx, event in enumerate(context.decoded_events):
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == from_address and
+                event.asset == self.eth and
+                event.address == L1_MESSENGER_PROXY
+            ):
+                refund_event_idx = idx
+                refund_amount = event.balance.amount
+                break
+        if refund_event_idx is not None:
+            del context.decoded_events[refund_event_idx]
+
+        # Find the corresponding transfer event and update it
+        for event in context.decoded_events:
+            if (
+                event.event_type == expected_event_type and
+                event.asset == asset and
+                event.balance.amount == amount
+            ):
+                bridge_match_transfer(
+                    event=event,
+                    from_address=from_address,
+                    to_address=to_address,
+                    from_chain=from_chain,
+                    to_chain=to_chain,
+                    amount=amount,
+                    asset=asset,
+                    expected_event_type=expected_event_type,
+                    new_event_type=new_event_type,
+                    counterparty=SCROLL_CPT_DETAILS,
+                )
+            elif (
+                event.event_type == HistoryEventType.SPEND and
+                event.asset == self.eth and
+                event.address == L1_GATEWAY_ROUTER
+            ):
+                l2_fees_amount = event.balance.amount - refund_amount
+                event.counterparty = CPT_SCROLL
+                event.balance.amount = l2_fees_amount
+                event.notes = (
+                    f'Spend {l2_fees_amount} ETH in L2 fees to bridge ERC20 tokens to Scroll'
+                )
```

Comment:
in general in those cases we prefer to use # noqa: E501 rathern than creating the string with this syntax